### PR TITLE
Repo name added to automatically generated commit message when mergin…

### PR DESCRIPTION
**Pull request by shashvat-kedia**
Fix for issue #9898.

### CODE REVIEW

File: models/pull.go [hunk 1]
```diff
@@ -185,7 +185,7 @@ func (pr *PullRequest) GetDefaultMergeMessage() string {
 		log.Error("Cannot load issue %d for PR id %d: Error: %v", pr.IssueID, pr.ID, err)
 		return ""
 	}
-	return fmt.Sprintf("Merge pull request '%s' (#%d) from %s/%s into %s", pr.Issue.Title, pr.Issue.Index, pr.MustHeadUserName(), pr.HeadBranch, pr.BaseBranch)
+	return fmt.Sprintf("Merge pull request '%s' (#%d) from %s/%s/%s into %s", pr.Issue.Title, pr.Issue.Index, pr.MustHeadUserName(), pr.HeadRepo.Name, pr.HeadBranch, pr.BaseBranch)
```
Comment (zeripath) [inline models/pull.go]: If prefer it if the delimiter between branch and repo was a colon or a space not a slash.

```suggestion
	return fmt.Sprintf("Merge pull request '%s' (#%d) from %s/%s:%s into %s", pr.Issue.Title, pr.Issue.Index, pr.MustHeadUserName(), pr.HeadRepo.Name, pr.HeadBranch, pr.BaseBranch)
```

That way it's clear that it's not a repo name.

**Review by zeripath — APPROVED** [review]
Small nit but otherwise approve.

**Comment by zeripath** [comment]
The current behaviour is actually completely specified as you can't have more than one fork of a repo and can't pr from something that isn't a fork.

However I can see that it is confusing.

If we're going to go for fully specified repo names then we should have a different separator between branch and repo name. Either colon or space. 

Whatever we choose it should then propagate through to the compare page. 

At present I think that uses a colon so colon is recommendeded

File: models/pull.go (hunk 1 above)
Comment (shashvat-kedia) [reply models/pull.go]: @zeripath I have made the change.

File: models/pull.go [hunk 2]
```diff
@@ -185,7 +185,7 @@ func (pr *PullRequest) GetDefaultMergeMessage() string {
 		log.Error("Cannot load issue %d for PR id %d: Error: %v", pr.IssueID, pr.ID, err)
 		return ""
 	}
-	return fmt.Sprintf("Merge pull request '%s' (#%d) from %s/%s into %s", pr.Issue.Title, pr.Issue.Index, pr.MustHeadUserName(), pr.HeadBranch, pr.BaseBranch)
+	return fmt.Sprintf("Merge pull request '%s' (#%d) from %s/%s:%s into %s", pr.Issue.Title, pr.Issue.Index, pr.MustHeadUserName(), pr.HeadRepo.Name, pr.HeadBranch, pr.BaseBranch)
```
Comment (zeripath) [inline models/pull.go]: Actually thinking on it should be:

```suggestion
	return fmt.Sprintf("Merge pull request '%s' (#%d) from %s:%s into %s", pr.Issue.Title, pr.Issue.Index, pr.HeadRepo.FullName(), pr.HeadBranch, pr.BaseBranch)
```

**Comment by zeripath** [comment]
Actually we should use FullName() for this. (That might make backporting a bit of a problem as we've made a small improvement in master/1.12-dev whereby the repo.OwnerName is stored on the repo table.

File: models/pull.go (hunk 2 above)
Comment (shashvat-kedia) [reply models/pull.go]: @zeripath I have made the change.

**Review by lafriks — APPROVED** [review]

**Review by adelowo — APPROVED** [review]

File: models/pull.go [hunk 3]
```diff
@@ -185,7 +185,7 @@ func (pr *PullRequest) GetDefaultMergeMessage() string {
 		log.Error("Cannot load issue %d for PR id %d: Error: %v", pr.IssueID, pr.ID, err)
```
Comment (lunny) [inline models/pull.go:185]: I would like to confirm if the head repository is the same as the base repository. Then return different message. When that is the same repository, we could ignore `%s/%s:`

**Review by lunny — DISMISSED** [review]

File: models/pull.go (hunk 2 above)
Comment (zeripath) [reply models/pull.go]: Ah you didn't quite make that correctly but I've fixed it now.

File: models/pull.go (hunk 3 above)
Comment (zeripath) [reply models/pull.go:185]: @lunny take a look I've just pushed some changes up.

**Review by 6543 — APPROVED** [review]

**Comment by zeripath** [comment]
Please send backport

**Pull request merged**